__tests__/models.test.js

```diff
@@ -99,6 +99,133 @@ describe("GET /api/reviews", () => {
         expect(shouldBe0.comment_count).toBe("0");
       });
   });
+  describe("?query", () => {
+    describe("category", () => {
+      test("200: features a category query, that selects the reviews by the specified category", () => {
+        return request(app)
+          .get("/api/reviews?category=social deduction")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toHaveLength(11);
+            reviews.forEach((review) => {
+              expect(review).toHaveProperty("category", "social deduction");
+            });
+          });
+      });
+      test("200: endpoint responds with all the reviews, when query is omitted", () => {
+        return request(app)
+          .get("/api/reviews")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toHaveLength(13);
+          });
+      });
+      //is this necessary, as it is done in the original get request right?
+      describe("Error Handling:", () => {
+        test("400: category doesn't exist", () => {
+          return request(app)
+            .get("/api/reviews?category=card game")
+            .expect(400)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe("Bad Request: Category does not exist!");
+            });
+        });
+        test("200: category exists but has no reviews", () => {
+          return request(app)
+            .get("/api/reviews?category=children's games")
+            .expect(200)
+            .then(({ body: { reviews } }) => {
+              expect(reviews).toEqual([]);
+            });
+        });
+        // autopasses as the code doesn't allow for incorrect categories, thus never needs promise reject at row count 0
+      });
+    });
+    describe("sort_by", () => {
+      test("200: reviews can be sorted by different columns via query", () => {
+        return request(app)
+          .get("/api/reviews?sort_by=title")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toBeSortedBy("title", { descending: true });
+          });
+      });
+      test("200: reviews are sorted by date by default", () => {
+        return request(app)
+          .get("/api/reviews")
+          .expect(200)
+          .then(({ body: { reviews } }) => {
+            expect(reviews).toBeSortedBy("created_at", { descending: true });
+          });
+      });
```

Comment:
This test is also not needed as it would have been tested in the original get request